src/Network/Mattermost/Connection.hs

```diff
@@ -102,7 +102,7 @@ submitRequest :: ConnectionData
               -> B.ByteString
               -> IO HTTP.Response_String
 submitRequest cd mToken method uri payload = do
-  path <- mmPath ("/api/v4" ++ uri)
+  path <- mmPath (cdDirectory cd ++ "/api/v4" ++ uri)
```

Comment:
This change assumes the `cdDirectory` starts with a slash. It will probably need to handle the case where a leading slash is missing.

Reply:
Would it be reasonable to just append the leading slash in `mkConnectionData`?

Another thing regarding slashes that I came across while testing: if there are extra slashes, the server may send a 302 redirecting to the right path. This causes requests to fail if a user specifies the path as `/mattermost/` instead of `/mattermost`. This doesn't affect the Mattermost desktop application, since it just follows the redirect.

Reply:
For the multiple slashes issue, the typical way to handle that is to construct a URI, which implicitly normalizes those slashes away, and then print the URI. For that, see https://hackage.haskell.org/package/network-uri. For this and other reasons, manual string construction of URIs can be error-prone.

Reply:
Even so, if we properly normalize the URI and get the path from it, we could still get a 302 response. Handling that is a separate but valid issue.

Reply:
That library does not appear to get rid of extra slashes. For example, the URI parsed from `http://example.com/a////b` still prints with all the extra slashes, and calling `pathSegments` on that returns `["a","","","","b"]`.

There is also a function called `normalizePathSegments`, but that only deals with `.` and `..` in paths.

Reply:
Ah, I see. I didn't realize it wasn't automatic in that package. This package claims to handle it, though: https://github.com/mrkkrp/modern-uri